### DISCUSSION TITLE
Comparitive rusults of using a custom depth map vs built-in depth map generation.

### DISCUSSION THREAD

**Discussion by bruibbruib** [Mikubill/sd-webui-controlnet#202] [Show and tell]
The new depth_leres mode gets more detail than standard depth mode, but using a custom depth map (from the sd Depthmap script, tweaked in photoshop and exported without transparency) got the result I needed in this case. 
The custom depthmap keeps the face closer to the original: It is still the same cat. But the built-in methods wander a bit more, and the cats personage varies too much. It is subtle, and in many cases would not matter, but the difference in these images means I can complete a project. Without the custom map I had been struggling.
What a fantastic tool this is!!
![jyhgjhgfj](https://user-images.githubusercontent.com/82312962/219885042-7baf1146-68df-4cf9-8651-d72f60588a2a.png)

![hgjhgfddjhg](https://user-images.githubusercontent.com/82312962/219885048-75c399f8-a18a-4d06-81cc-9de620849b12.png)

![hbgfdhgdfjf](https://user-images.githubusercontent.com/82312962/219885049-81e98a91-6141-4c89-90b3-2ede2a9d4a96.png)

**Comment by irakli-ff** [comment]
@bruibbruib can you explain how to use custom depth map in Controlnet workflow? thanks!

**Reply by bruibbruib** [reply]
load in the depth map into the CN, set pre-processor to 'none', set Model to depth. canvas width, etc doesn't matter. Mine was not working for a while as the .PNG I was using had transparency, resaved .PNG without transparency and it worked.
See attached pic:
![kjhkjgk](https://user-images.githubusercontent.com/82312962/221437632-9db6ba5c-1f76-40ee-aa92-5aabe3f34d98.PNG)

**Comment by marcsyp** [comment]
Interested in this, as I have found the depth maps generated by CN to also be a bit flaky, even though can be pretty great.  Can you explain more about the "SD depth map" script and the tweaks you make in Photoshop, and for what reason they need tweaking?  I am not familiar with the SD depth map script, I don't see it in my standard list....

Thanks!

**Reply by bruibbruib** [reply]
Oops! missed the reply box. see my answer below!

**Comment by bruibbruib** [comment]
This is the script/extension I used for depth maps generation: 
https://github.com/thygate/stable-diffusion-webui-depthmap-script
I use this to get some pretty nice maps, but the output files were not directly readable by CN, so I took them to photoshop just to re-save the file, just without transparency. Then the files are read ok by CN.

Also in photoshop, I will use the levels tool to maximise the usage of the greyscale gradient. The first image is how a depth map looks rendered from C4D. Nearly black! Second image is after levels.

![hgfhgfsdh](https://user-images.githubusercontent.com/82312962/221582933-09814b4f-720a-4d53-abba-07f2dde8a311.PNG)

![njhgnhgdfhj](https://user-images.githubusercontent.com/82312962/221582965-39b978ce-ab7a-449a-bd44-fabb3dbd74ea.PNG)

I'm just fumbling around here, trying to find what works!!
I hope this helps.
